src/zenml/integrations/github/orchestrators/github_actions_orchestrator.py

```diff
@@ -0,0 +1,460 @@
+        append_secret_placeholder = (
+            "echo {secret_name}=${{{{ secrets.{secret_name} }}}} >> {file}; "
+        )
+        for secret_name in secrets_manager.get_all_secret_keys(
+            include_prefix=True
+        ):
+            command += append_secret_placeholder.format(
+                secret_name=secret_name, file=file_name
+            )
+
+        return {
+            "name": "Write environment file",
+            "run": command,
+        }
+
+    def prepare_pipeline_deployment(
+        self,
+        pipeline: "BasePipeline",
+        stack: "Stack",
+        runtime_configuration: "RuntimeConfiguration",
+    ) -> None:
+        """Builds a docker image for the current environment and uploads it to
+        a container registry if configured.
+        """
+        if self.prevent_dirty_repository and self.git_repo.is_dirty(
+            untracked_files=True
+        ):
+            raise RuntimeError(
+                "Trying to run a pipeline from within a dirty git repository."
+                "If you want this orchestrator to skip the dirty repo check in "
+                f"the future, run\n `zenml orchestrator update {self.name} "
+                "--prevent_dirty_repository=false`"
+            )
+
+        image_name = self.get_docker_image_name(pipeline.name)
+        requirements = {*stack.requirements(), *pipeline.requirements}
+
+        logger.debug(
+            "Github actions docker image requirements: %s", requirements
+        )
+
+        docker_utils.build_docker_image(
+            build_context_path=source_utils.get_source_root_path(),
+            image_name=image_name,
+            dockerignore_path=pipeline.dockerignore_file,
+            requirements=requirements,
+            base_image=self.custom_docker_base_image_name,
+        )
+
+        assert stack.container_registry  # should never happen due to validation
+        stack.container_registry.push_image(image_name)
+
+        # Store the docker image digest in the runtime configuration so it gets
+        # tracked in the ZenStore
+        image_digest = docker_utils.get_image_digest(image_name) or image_name
+        runtime_configuration["docker_image"] = image_digest
+
+    def prepare_or_run_pipeline(
+        self,
+        sorted_steps: List[BaseStep],
+        pipeline: "BasePipeline",
+        pb2_pipeline: Pb2Pipeline,
+        stack: "Stack",
+        runtime_configuration: "RuntimeConfiguration",
+    ) -> Any:
+        """Writes a GitHub Action workflow yaml and optionally pushes it."""
+        workflow_path = os.path.join(
+            self.workflow_directory, f"{pipeline.name}.yaml"
+        )
+
+        workflow_dict: Dict[str, Any] = {"name": pipeline.name}
+
+        schedule = runtime_configuration.schedule
+        if schedule:
+            if not schedule.cron_expression:
+                raise ValueError(
+                    "GitHub Action workflows can only be scheduled using cron "
```

Comment:
In the long run we could write a converter from interval to cron to make it possible no matter what